src/commands/deploy.ts

```diff
@@ -314,6 +321,25 @@ function getRecommendedTaskName(language: ProjectLanguage, runtime: ProjectRunti
     }
 }
 
+async function promptToBuildNativeDeps(actionContext: IActionContext, deployFsPath: string, scmType: string | undefined): Promise<appservice.IPreDeployTaskResult> {
+    const flag: string = '--build-native-deps';
+    const message: string = localize('funcPackFailed', 'Failed to package your project. Use a Docker container to build incompatible dependencies?', flag);
+    const result: vscode.MessageItem | undefined = await vscode.window.showErrorMessage(message, { modal: true }, DialogResponses.yes, DialogResponses.learnMore);
+    if (result === DialogResponses.yes) {
+        actionContext.properties.preDeployTaskResponse = 'packNativeDeps';
+        await updateWorkspaceSetting(preDeployTaskSetting, `${packTaskName} ${flag}`, deployFsPath);
+        return await appservice.tryRunPreDeployTask(actionContext, deployFsPath, scmType, extensionPrefix);
+    } else if (result === DialogResponses.learnMore) {
```

Comment:
Should `Learn More` really close the information window?  I think it'd be annoying to learn more about the problem, but then have to recreate the problem for the prompt to reappear.

Reply:
Two things:
1. I don't think it's clear cut for modal dialogs. They prevent you from doing anything else in VS Code so I could see it being annoying if we kept popping it up.
1. I think the while loop makes the code look really ugly when we do that. I would leverage shared code if we had it, but I don't want to re-implement that every time